### ISSUE TITLE
[Request] Export activity as CSV

### ISSUE THREAD

**Issue by crankygeek** [ProxymanApp/Proxyman#383]
I would like to be able to export the currently displayed activity as a CSV for evaluation in outside tools, such as a spreadsheet.

**Comment by shirshak55** [comment]
@NghiaTranUIT I think there are many more other option a plugin system just like sublime text would be better?

**Comment by crankygeek** [comment]
Ran into a need for this again last week. I'd be happy with the ability to select and copy the table contents, and handle pasting/whatever myself without any formal export.

**Comment by julianozen** [comment]
+1 for this. this would be a huge win if we could copy the results as a CSV or similar and paste into Google Sheets / Excel. Charles offers this functionality

**Comment by NghiaTranUIT** [comment]
Thanks for your upvote. Since it's straightforward to implement, we will ship it in the next two releases 👍

**Comment by julianozen** [comment]
OMG such fast service. Thank you

**Comment by shirshak55** [comment]
@julianozen sarcasm ?

**Comment by crankygeek** [comment]
@NghiaTranUIT Just to be clear, copying the current results table to paste into a spreadsheet is still work in progress?

**Comment by NghiaTranUIT** [comment]
Yes, it's. It will be implemented in 2.29.0 👍 

Sorry for the delay because there are a lot of waiting features 😢

**Comment by crankygeek** [comment]
No worries. Just confirming since I didn't see it.

Too many waiting features is a sign of success! :)

**Comment by julianozen** [comment]
@shirshak55 No! No sarcasm at all. I came across this post when searching for a way to export to CSV. I decided to add a +1 for the feature request and got a response in less then 1 minutes for @NghiaTranUIT saying it would be added in the next release. 

Like magic!

**Comment by julianozen** [comment]
@NghiaTranUIT If its not too much trouble, would it be possible to get access to 2.29.0 a little early (this would save me a lot of time with my current project at work 😅 ). If its a big hassle, no worries

**Comment by NghiaTranUIT** [comment]
Yes, I'm working on this feature. I will send you a BETA build this week 👍

**Comment by julianozen** [comment]
@NghiaTranUIT You are awesome

**Comment by NghiaTranUIT** [comment]
Hey @julianozen @crankygeek @shirshak55, please check out this Beta build: https://proxyman.s3.us-east-2.amazonaws.com/beta/Proxyman_2.28.0_Export_As_CSV.dmg

From now, you can export selected requests/responses to the CSV file.

- CSV contains all essential fields like the Charles Proxy app
- It also contains the [Custom Header Column](https://docs.proxyman.io/basic-features/custom-header-column) too, so you can export some important Header Key-Value

Please let me know if there is any data you need in the CSV, I will add it 👍 

### Screenshot
<img width="1055" alt="Screen Shot 2021-06-24 at 14 06 12" src="https://user-images.githubusercontent.com/5878421/123218478-f789a080-d4f5-11eb-8a56-8f71dd5ff8f9.png">
<img width="2672" alt="Screen Shot 2021-06-24 at 13 02 00" src="https://user-images.githubusercontent.com/5878421/123218521-053f2600-d4f6-11eb-8282-e1ad2073104f.png">
<img width="2672" alt="Screen_Shot_2021-06-24_at_13_02_51" src="https://user-images.githubusercontent.com/5878421/123218587-1b4ce680-d4f6-11eb-9e1d-d55336532672.png">

**Comment by crankygeek** [comment]
@NghiaTranUIT This looks great!

I think I'd prefer if the export matched my current columns and column order, but I'm also okay with modifying the data in my spreadsheet app.

Only thing that seems to be missing is the Edit column. I can't think of any reason that I need it myself, but I could see it being useful for some folks, e.g. "Why is this request so out of whack with everything else in the data set, oh, right, that's the one I modified" kind of thing.

**Comment by julianozen** [comment]
This rocks. This works as I was hoping for. 

<img width="1680" alt="Screen Shot 2021-06-24 at 11 11 36 AM" src="https://user-images.githubusercontent.com/3589488/123312268-f39d5480-d4dc-11eb-8fd8-46f1bc9edfa1.png">

Now I plan to do this over about 30 screens to monitor the endpoints and duration of the requests

**Comment by NghiaTranUIT** [comment]
Thanks for letting me know it work 👍 

It's a good idea to provide an ability to edit the CSV columns @crankygeek, I will collect more requests and might implement it soon 👍
